### PR TITLE
fix: file not found errors when the root has a trailing slash

### PR DESCRIPTION
**Pull request by gurgunday**
It currently does not handle a root with a trailing slash correctly

### CODE REVIEW

**Comment by gurgunday** [comment]
Have no idea how to test this...

It's not working on production, which runs on Linux, and I don't have a Linux machine right now

File: index.js [hunk 1]
```diff
@@ -381,7 +383,8 @@ async function fastifyStatic (fastify, opts) {
 
   function setUpHeadAndGet (routeOpts, route, file, rootPath) {
     const toSetUp = Object.assign({}, routeOpts, {
-      method: ['HEAD', 'GET'],
+      exposeHeadRoute: true,
+      method: 'GET',
```
Comment (climba03003) [inline index.js:387]: Explicitly adding `HEAD` is an optimization.

**Review by climba03003 — COMMENTED** [review]
I would like to see a test before merging.

If it is not available, you should try to provide comparison on the result between `v8` and `v10` of `glob`.

**Comment by gurgunday** [comment]
I really don’t know where the issue is since linux tests are passing too…

I’ll try to see if I can do something about it

**Comment by gurgunday** [comment]
Alright, I got the issue, will send a test now

File: index.js [hunk 2]
```diff
@@ -112,33 +111,30 @@ async function fastifyStatic (fastify, opts) {
...
-      const globPattern = '**/**'
 
       const roots = Array.isArray(sendOptions.root) ? sendOptions.root : [sendOptions.root]
-      for (let i = 0; i < roots.length; ++i) {
-        const rootPath = roots[i]
-        const posixRootPath = rootPath.split(path.win32.sep).join(path.posix.sep)
-        const files = await glob(`${posixRootPath}/${globPattern}`, { follow: true, nodir: true, dot: opts.serveDotFiles })
+      for (let rootPath of roots) {
+        rootPath = rootPath.split(path.win32.sep).join(path.posix.sep)
+        !rootPath.endsWith('/') && (rootPath += '/')
```
Comment (Eomm) [inline index.js:130]:
```suggestion
        if(!rootPath.endsWith(path.posix.sep)){
          rootPath += path.posix.sep
        }
```

Comment (gurgunday) [reply index.js:130]: Istanbul complains

**Comment by gurgunday** [comment]
I really need to make this release, so my apologies for merging without an approval, please do let me of your feedback and I'll make a second PR addressing them

**Review by Uzlopak — COMMENTED** [review]
retrospective indemnity approval

**Pull request merged**